### PR TITLE
Fix reset interactive build cancellation

### PR DESCRIPTION
**Pull request by drognanar**
Do not call CancelBuildProject after build has already completed.

### CODE REVIEW

**Comment by drognanar** [comment]
FYI @dotnet/roslyn-interactive

File: src/Interactive/EditorFeatures/Core/Extensibility/Interactive/ResetInteractive.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ internal Task Execute(IInteractiveWindow interactiveWindow, string title)
 
             // If the user hits the cancel button on the wait indicator, then we want to stop the
             // build.
-            waitContext.CancellationToken.Register(() =>
+            var buildCancellation = waitContext.CancellationToken.Register(() =>
```
Comment (KevinH-MS) [inline src/Interactive/EditorFeatures/Core/Extensibility/Interactive/ResetInteractive.cs]: using or try/finally in case we throw?

Comment (drognanar) [inline src/Interactive/EditorFeatures/Core/Extensibility/Interactive/ResetInteractive.cs]: Thanks, I should have thought of that. Made the relevant change.

**Comment by KevinH-MS** [comment]
Would be nice to have a test if at all possible, but code change LGTM :+1:

**Comment by ManishJayaswal** [comment]
:+1:

**Comment by Pilchie** [comment]
Approved, but please talk to @davkean before merging, as we are still trying to track down a perf regression and want to keep the branch from churning.

**Pull request merged**